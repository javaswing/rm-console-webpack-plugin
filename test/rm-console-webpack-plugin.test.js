
import RmConsoleWebpackPlugin from "../src/index";

import { compile, getCompiler, cleanErrorStack } from "./helpers/index";

describe("RmConsoleWebpackPlugin", () => {
  it("test rm console", () => {
    const compiler = getCompiler();

    new RmConsoleWebpackPlugin({}).apply(compiler);
    return compile(compiler).then(({ stats }) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot("errors");
      expect(warnings).toMatchSnapshot("warnings");

      for (const file in stats.compilation.assets) {
        if (
          Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)
        ) {
          console.log('file', stats.compilation.assets[file])
          expect(stats.compilation.assets[file].size()).toMatchSnapshot(file);
        }
      }
    });
  });
});
