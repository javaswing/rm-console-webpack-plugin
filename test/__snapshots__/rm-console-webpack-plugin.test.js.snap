// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RmConsoleWebpackPlugin test rm console: errors 1`] = `[]`;

exports[`RmConsoleWebpackPlugin test rm console: main.510eee11fbcff201957a.js 1`] = `4675`;

exports[`RmConsoleWebpackPlugin test rm console: warnings 1`] = `[]`;
